devtools/x/src/diff_summary.rs

```diff
@@ -23,6 +26,12 @@ pub fn run(args: Args, _xctx: XContext) -> crate::Result<()> {
     let compare_summary = Summary::parse(&compare_summary_text)?;
 
     let summary_diff = SummaryDiff::new(&base_summary, &compare_summary);
-    println!("{}", summary_diff.report());
+
+    match args.output_format.as_deref() {
+        Some("json") => println!("{}", serde_json::to_string(&summary_diff)?),
+        Some("toml") => println!("{}", toml::to_string(&summary_diff)?),
+        _ => println!("{}", summary_diff.report()),
```

Comment:
you should accept "text" or nothing but reject all other options :)

```suggestion
        Some("text") | None => println!("{}", summary_diff.report()),
        _ => return Err("--output-format only accepts json, html or text"),
```